x-pack/legacy/plugins/rollup/index.ts

```diff
@@ -45,22 +38,31 @@ export function rollup(kibana) {
       visualize: ['plugins/rollup/legacy'],
       search: ['plugins/rollup/legacy'],
     },
-    init: function(server) {
-      const { usageCollection } = server.newPlatform.setup.plugins;
-      registerLicenseChecker(server);
-      registerIndicesRoute(server);
-      registerFieldsForWildcardRoute(server);
-      registerSearchRoute(server);
-      registerJobsRoute(server);
-      registerRollupUsageCollector(usageCollection, server);
-      if (
-        server.plugins.index_management &&
-        server.plugins.index_management.addIndexManagementDataEnricher
-      ) {
-        server.plugins.index_management.addIndexManagementDataEnricher(rollupDataEnricher);
-      }
+    init(server: any) {
+      const { core: coreSetup, plugins } = server.newPlatform.setup;
+      const { usageCollection, metrics } = plugins;
+
+      const rollupSetup = (plugins.rollup as unknown) as RollupSetup;
 
-      registerRollupSearchStrategy(this.kbnServer);
+      const initContext = ({
+        config: rollupSetup.__legacy.config,
+        logger: rollupSetup.__legacy.logger,
+      } as unknown) as PluginInitializerContext;
+
+      const rollupPluginInstance = plugin(initContext);
+
+      rollupPluginInstance.setup(coreSetup, {
+        usageCollection,
+        metrics,
+        __LEGACY: {
+          route: server.route.bind(server),
```

Comment:
Is this used anywhere? couldn't find a reference.

Reply:
Good catch. You're right, it's not needed anymore.